src/PAModel/Schemas/PcfControls/PcfControl.cs

```diff
@@ -0,0 +1,164 @@
+using Microsoft.PowerPlatform.Formulas.Tools.EditorState;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace Microsoft.PowerPlatform.Formulas.Tools.Schemas
+{
+    class PcfControl
+    {
+        private const string EventsKey = "Events";
+        private const string CommonEventsKey = "CommonEvents";
+        private const string ResourcesKey = "Resources";
+        private const string PropertiesKey = "Properties";
+        private const string NamespaceKey = "ControlNamespace";
+        private const string ConstructorKey = "ControlConstructor";
+        private const string IncludedPropertiesKey = "IncludedProperties";
+        private const string PropertyDependenciesKey = "PropertyDependencies";
+        private const string SubscribedFunctionalitiesKey = "SubscribedFunctionalities";
+        private const string AuthConfigPropertiesKey = "AuthConfigProperties";
+        private const string DataConnectorsKey = "DataConnectors";
+
+        public string ControlNamespace { get; set; }
+        public string ControlConstructor { get; set; }
+        public string FullyQualifiedControlName { get; set; }
+        public Resource[] Resources { get; set; }
+        public IDictionary<string, string> SubscribedFunctionalities { get; set; }
+        public IEnumerable<Property> Properties { get; set; }
+        public IEnumerable<Property> IncludedProperties { get; set; }
+        public IEnumerable<IDictionary<string, AuthConfigProperty>> AuthConfigProperties { get; set; }
+        public IEnumerable<PropertyDependency> PropertyDependencies { get; set; }
+        public IEnumerable<DataConnectorMetadata> DataConnectors { get; set; }
+        public IEnumerable<Event> Events { get; set; }
+        public IEnumerable<Event> CommonEvents { get; set; }
+
+        public static PcfControl GetPowerAppsControlFromJson(CombinedTemplateState template)
+        {
+            var powerAppControl = new PcfControl();
+
+            var dynamicControlDefinition = Utilities.JsonParse<IDictionary<string, string>>(template.DynamicControlDefinitionJson);
```

Comment:
This is interesting, is there a reason we couldn't parse this into an POCO instead?

Reply:
I borrowed this logic from doc server, also since dynamicControlDefinitionJson is doubly encoded it deserializing to class object might not work.